src/linker/Linker/CompilerGeneratedState.cs

```diff
@@ -75,6 +84,13 @@ void ProcessMethod (MethodDefinition method)
 						if (lambdaOrLocalFunction == null)
 							continue;
 
+						if (lambdaOrLocalFunction.IsConstructor && CompilerGeneratedNames.IsLambdaDisplayClass (lambdaOrLocalFunction.DeclaringType.Name)) {
```

Comment:
I think there has to be a call to a constructor (or at least initobj) if the display class is generic, even if it's a struct. Otherwise, there's no substitution. I'll try to construct a test case.

Reply:
Turns out, this is impossible in the current scope because there are no struct display classes for iterators or async methods.